ott/core/icnn.py

```diff
@@ -88,66 +43,142 @@ class ICNN(nn.Module):
       `jax.nn.initializers.normal`).
     act_fn: choice of activation function used in network architecture
       (needs to be convex, default: `nn.leaky_relu`).
+    pos_weights: choice to enforce positivity of weight or use regularizer.
+    gaussian_maps: data inputs of source and target measures for
+      initialization scheme based on Gaussian approximation of input and
+      target measure (if None, identity initialization is used).
   """
 
   dim_hidden: Sequence[int]
-  init_std: float = 0.1
+  init_std: float = 1e-1
   init_fn: Callable = jax.nn.initializers.normal
-  act_fn: Callable = nn.leaky_relu
+  act_fn: Callable = nn.relu
   pos_weights: bool = True
+  dim_data: int = 2
+  gaussian_map: Tuple = None
 
   def setup(self):
-    num_hidden = len(self.dim_hidden)
-
-    w_zs = []
+    self.num_hidden = len(self.dim_hidden)
 
     if self.pos_weights:
-      Dense = PositiveDense
+      hid_dense = PositiveDense
+      # this function needs to be the inverse map of function
+      # used in PositiveDense layers
+      rescale = lambda x: jnp.log(jnp.exp(x) - 1)
```

Comment:
I think this should come from (and be encapsulated within) the PositiveDense layer. Here is what I would do: 
- add in `PositiveDense` a function called `rectifier_fn`, by default the softplus.
- add also `inv_rectifier_fn`, by default the sofplusinv.
- add a simple assert like `inv_rectifier_fn(rectifier_fn(0.1))==0.1`, "Make sure both rectifier and inverse are defined properly"
- remove `beta` from `PositiveDense` (this can be defined by the user when passing a different rectifier)
- define `rescale = hid_dense.inv_rectifier_fn` in the code above.

Reply:
Done.